### PR TITLE
issue #39 - SQLite support

### PR DESCRIPTION
**Pull request by pounard**
This is a unfinished draft, some tests are failing due to the lack of `MD5()` or any other hash function in SQLite.

### CODE REVIEW

File: docs/content/anonymization/core-anonymizers.md [hunk 1]
```diff
@@ -6,6 +6,15 @@ This page list all *Anonymizers* provided by *DbToolsBundle*.
 
 ## EmailAnonymizer
 
+EmailAnonymizer uses a hash function on the original value to make each unique email
+anonymization reproducible accross tables.
+
+:::warning
+SQLite does implement `MD5()` function, neither any hashing function: in order to get
+around this, the `rowid` value is used instead which prevent email values anonymization
```
Comment (SimonMellerin) [inline docs/content/anonymization/core-anonymizers.md:14]: :+1:

File: docs/content/database-vendors.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ Here is a matrix of the current state of support:
 | MySQL 5.7 or higher    |  <Badge type="tip" text="✔" title="Working"/> | <Badge type="tip" text="✔" title="Working"/> | <Badge type="tip" text="✔" title="Working"/> |
 | SQLite    |  <Badge type="danger" text="✘" title="Unsupported"/> | <Badge type="warning" text="~" title="Only unit-tested"/> | <Badge type="danger" text="✘" title="Unsupported"/> |
 | Oracle    |  <Badge type="danger" text="✘" title="Unsupported"/> | <Badge type="warning" text="~" title="Only unit-tested"/>  | <Badge type="danger" text="✘" title="Unsupported"/> |
+| SQLite    |  <Badge type="danger" text="✘" title="Unsupported"/> | <Badge type="warning" text="~" title="Only unit-tested"/>  | <Badge type="danger" text="✘" title="Unsupported"/> |
```
Comment (SimonMellerin) [inline docs/content/database-vendors.md:27]: It was there already ;)

If you modify this part, remove Oracle which should not be there

File: tests/Functional/Anonymizer/Core/FloatAnonymizerTest.php [hunk 3]
```diff
@@ -44,6 +45,8 @@ protected function createTestData(): void
 
     public function testAnonymize(): void
     {
+        $this->skipIfDatabase(AbstractBridge::SERVER_SQLITE, 'SQLite RANDOM() return an int, not a float, requires specific code.');
```
Comment (SimonMellerin) [inline tests/Functional/Anonymizer/Core/FloatAnonymizerTest.php:48]: I don't understand the problem here, this anonymizer use ::getRandomExpression which return an integer, then the expresion is 
```
$expr->raw(
                    'FLOOR(? * (? - ? + 1) + ?) / ?',
                    [
                        $this->getRandomExpression(),
                        $expr->cast($max * $precision, 'int'),
                        $min * $precision,
                        $min * $precision,
                        $precision
                    ]
                )
```

This should work ?

**Review by SimonMellerin — CHANGES_REQUESTED** [review]

File: tests/Functional/Anonymizer/Core/FloatAnonymizerTest.php (hunk 3 above)
Comment (pounard) [reply tests/Functional/Anonymizer/Core/FloatAnonymizerTest.php:48]: Hum, with randomInt() yes, it might do the trick !

**Review by SimonMellerin — APPROVED** [review]

**Pull request merged**